### PR TITLE
DEPR: remove itemsize, data, base, flags, strides

### PR DESCRIPTION
**Pull request by jbrockmendel**
and un-deprecated `item`, though im not sure where that goes in the whatsnew

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -693,49 +684,14 @@ def item(self):
...
-        """
-        warnings.warn(
-            "{obj}.itemsize is deprecated and will be removed "
-            "in a future version".format(obj=type(self).__name__),
-            FutureWarning,
-            stacklevel=2,
-        )
-        return self._ndarray_values.itemsize
+        if len(self) == 1:
+            return next(iter(self))
```
Comment (jreback) [inline pandas/core/base.py]: so we have sufficient tests that this preserves the correct type
eg  Timestamp with tz 

i think it does but should test

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jorisvandenbossche** [comment]
You will need to remove some things from the API docs (that's what the doc failure is complaining about)

**Comment by jbrockmendel** [comment]
Will update per comments.  Also need to change the implementation so item() raises for len!=1

**Comment by jorisvandenbossche** [comment]
Can you leave the `item` change for a separate PR? 
That change deserves specific attention I think, not hidden in an "easy remove some stuff" PR

**Comment by jbrockmendel** [comment]
> Can you leave the item change for a separate PR?

Sounds good

**Comment by jbrockmendel** [comment]
updated+green, un-doing `item` changes

**Comment by jreback** [comment]
lgtm, need a rebase

**Review by jreback — APPROVED** [review]

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**